src/mesh/mesh.cpp

```diff
@@ -1084,24 +1084,58 @@ void Mesh::Initialize(bool init_problem, ParameterInput *pin, ApplicationInput *
     }
 
     // send FillGhost variables
+    bool can_delete;
+    int test_iters = 0;
+    do {
+      can_delete = true;
+      for (auto &[k, comm] : boundary_comm_map) {
+        can_delete = comm.IsSafeToDelete() && can_delete;
+      }
+      test_iters++;
+    } while (!can_delete && test_iters < 1e10);
+    if (test_iters >= 1e10)
+      PARTHENON_FAIL(
+          "Too many iterations waiting to delete boundary communication buffers.");
```

Comment:
`PARTHENON_REQUIRE` instead of `if ... PARTHENON_FAIL`?

Reply:
changed.